### PR TITLE
create showLoading prop and erase unused vars/state

### PR DESCRIPTION
**Pull request by jgfidelis**
https://testestore--storecomponents.myvtex.com/

Eu quero a partir do componente pai fazer ele entrar num estado de loading (aguardando outro processamento por exemplo).

Não sei se `showLoading` é o melhor nome rs

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## [Unreleased]
 
 ## [3.16.1] - 2019-02-13
+### Changed
+- Remove unused state and vars on `BuyButton`
+- Create showLoading prop for `BuyButton`
```
Comment (klzns) [inline CHANGELOG.md]: ⛏ 
```suggestion
- Create `showLoading` prop for `BuyButton`
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## [Unreleased]
 
 ## [3.16.1] - 2019-02-13
```
Comment (klzns) [inline CHANGELOG.md]: Is the version already published?

Shouldn't it be inside unreleased?

File: react/components/BuyButton/index.js [hunk 3]
```diff
@@ -177,6 +173,8 @@ BuyButton.propTypes = {
   onAddFinish: PropTypes.func,
   /** Add to cart mutation */
   addToCart: PropTypes.func.isRequired,
+  /** If the button should show the loading spinner */
```
Comment (brunojdo) [inline react/components/BuyButton/index.js]: Show loading in the button

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
+- Remove unused state and vars on `BuyButton`
+- Create `showLoading` prop for `BuyButton`
```
Comment (brunojdo) [inline CHANGELOG.md]: I prefer `activeLoading` prop

**Pull request closed without merging**